cmake/Findllvm.cmake

```diff
@@ -6,177 +6,181 @@
 # LLVM_INCLUDE_DIRS
 # LLVM_LIBRARIES
 # LLVM_LIBDIRS
+# LLVM_LINK_MODE
 
-find_path(LLVM_INCLUDE_DIRS NAMES llvm/IR/IRBuilder.h
-  PATHS
-    /usr/lib/llvm/14/include
-    /usr/lib/llvm-14/include
-    /usr/lib/llvm-14.0/include
-    /usr/lib/llvm14/include
-    /usr/local/llvm14/include
-    /usr/local/llvm140/include
-    /usr/local/opt/llvm@14/include
-    /opt/homebrew/opt/llvm@14/include
-    /mingw64/include
-)
 
-if(ZIG_PREFER_CLANG_CPP_DYLIB)
-  find_library(LLVM_LIBRARIES
-    NAMES
-      LLVM-14.0
-      LLVM-14
-      LLVM-140
-      LLVM
-    NAMES_PER_DIR
-    PATHS
-      ${LLVM_LIBDIRS}
-      /usr/lib/llvm/14/lib
-      /usr/lib/llvm/14/lib64
-      /usr/lib/llvm-14/lib
-      /usr/lib/llvm14/lib
-      /usr/local/llvm14/lib
-      /usr/local/llvm140/lib
-      /usr/local/opt/llvm@14/lib
-      /opt/homebrew/opt/llvm@14/lib
-  )
+if(ZIG_USE_LLVM_CONFIG)
+  while(1)
+    unset(LLVM_CONFIG_EXE CACHE)
```

Comment:
What's the goal here, why unset LLVM_CONFIG_EXE, what if we want to `-DLLVM_CONFIG_EXE=<some/path/to/llvm-config>` ?